trial/src/Trial.hs

```diff
@@ -102,17 +125,29 @@ pattern Error <- E
 
 withW :: Functor f => f e -> f (Fatality, e)
 withW = fmap (W,)
+{-# INLINE withW #-}
 
-{- |
+{- | 'Trial' is a data type that stores history of all events happened
+with a value. In addition, each event is tagged with a value of type
```

Comment:
Tagged word is misleading in this context, let's use marked or just to rewrite the sentence a bit :)